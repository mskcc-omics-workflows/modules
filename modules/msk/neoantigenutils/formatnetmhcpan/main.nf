process NEOANTIGENUTILS_FORMATNETMHCPAN {
    tag "$meta.id"
    label 'process_single'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/neoantigen-utils-base:1.3.0':
        'docker.io/mskcc/neoantigen-utils-base:1.3.0' }"

    input:
    tuple val(meta),  path(netmhcPanOutput)

    output:
    tuple val(meta), path("*.tsv"),              emit: netMHCpanreformatted
    path "versions.yml",                         emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def netmhcOutputType = meta.typeMut ? "--type_MUT": ""
    def netmhcOutputFrom = meta.fromStab ? "--from_STAB": ""
    def netmhcOutputPan  = meta.fromPan ? "": "--from_NETMHC3"

    """
        format_netmhcpan_output.py \
            --netMHCpan_output ${netmhcPanOutput} \
            --id ${prefix} \
            ${netmhcOutputPan} \
            ${netmhcOutputType} \
            ${netmhcOutputFrom}

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            formatNetmhcpanOutput: \$(echo \$(format_netmhcpan_output.py -v))
        END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def netmhcOutputType = meta.typeMut ? "MUT": "WT"
    def netmhcOutputFrom = meta.fromStab ? "STAB": "PAN"
    """
        touch ${prefix}.${netmhcOutputType}.${netmhcOutputFrom}.tsv
        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            formatNetmhcpanOutput: \$(echo \$(format_netmhcpan_output.py -v))
        END_VERSIONS
    """
}
