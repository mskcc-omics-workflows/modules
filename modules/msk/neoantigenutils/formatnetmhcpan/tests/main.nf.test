nextflow_process {

    name "Test Process NEOANTIGENUTILS_FORMATNETMHCPAN"
    script "../main.nf"
    process "NEOANTIGENUTILS_FORMATNETMHCPAN"

    tag "modules"
    tag "modules_nfcore"
    tag "neoantigenutils"
    tag "neoantigenutils/formatnetmhcpan"
    tag "modules_msk"

    test("neoantigenutils_formatnetmhcpan - output(MUT,netmhcpan) - tsv") {

        when {

            process {
                """

                input[0] = [
                    [ id:'test', typeMut: true, fromStab: false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['MUTnetMHCpan'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("neoantigenutils_formatnetmhcpan - output(WT,netmhcpan) - tsv") {

        when {

            process {
                """

                input[0] = [
                    [ id:'test', typeMut: false, fromStab: false], // meta map
                    file(params.test_data_mskcc['neoantigen']['WTnetMHCpan'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

        test("neoantigenutils_formatnetmhcpan - output(MUT,netmhcpanstab) - tsv") {

        when {

            process {
                """

                input[0] = [
                    [ id:'test', typeMut: true, fromStab: true ], // meta map
                    file(params.test_data_mskcc['neoantigen']['MUTnetMHCpanstab'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("neoantigenutils_formatnetmhcpan - output(WT,netmhcpanstab) - tsv") {

        when {

            process {
                """

                input[0] = [
                    [ id:'test', typeMut: false, fromStab: false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['WTnetMHCpanstab'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

}
    test("neoantigenutils_formatnetmhcpan - output(MUT,netmhcpan) - tsv - stub") {

        options "-stub"

        when {

            process {
                """

                input[0] = [
                    [ id:'test', typeMut: true, fromStab: false ], // meta map
                    file('MUTnetMHCpan')
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
