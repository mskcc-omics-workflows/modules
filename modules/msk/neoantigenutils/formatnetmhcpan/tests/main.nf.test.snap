{
    "neoantigenutils_formatnetmhcpan - output(MUT,netmhcpan) - tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "typeMut": true,
                            "fromStab": false
                        },
                        "test_netmHCpanoutput.MUT.tsv:md5,7f00f2df190fe801700b626b72dfdb99"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2c02f5f3103ee1532c27f7f3b873a578"
                ],
                "netMHCpanreformatted": [
                    [
                        {
                            "id": "test",
                            "typeMut": true,
                            "fromStab": false
                        },
                        "test_netmHCpanoutput.MUT.tsv:md5,7f00f2df190fe801700b626b72dfdb99"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2c02f5f3103ee1532c27f7f3b873a578"
                ]
            }
        ],
        "timestamp": "2024-07-30T13:46:27.878268"
    },
    "neoantigenutils_formatnetmhcpan - output(MUT,netmhcpan) - tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "typeMut": true,
                            "fromStab": false
                        },
                        "test.MUT.PAN.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2c02f5f3103ee1532c27f7f3b873a578"
                ],
                "netMHCpanreformatted": [
                    [
                        {
                            "id": "test",
                            "typeMut": true,
                            "fromStab": false
                        },
                        "test.MUT.PAN.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2c02f5f3103ee1532c27f7f3b873a578"
                ]
            }
        ],
        "timestamp": "2024-07-30T13:47:05.72509"
    },
    "neoantigenutils_formatnetmhcpan - output(WT,netmhcpan) - tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "typeMut": false,
                            "fromStab": false
                        },
                        "test_netmHCpanoutput.WT.tsv:md5,a1d7db1b6f116e96457f2fa60660558e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2c02f5f3103ee1532c27f7f3b873a578"
                ],
                "netMHCpanreformatted": [
                    [
                        {
                            "id": "test",
                            "typeMut": false,
                            "fromStab": false
                        },
                        "test_netmHCpanoutput.WT.tsv:md5,a1d7db1b6f116e96457f2fa60660558e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2c02f5f3103ee1532c27f7f3b873a578"
                ]
            }
        ],
        "timestamp": "2024-07-30T13:46:37.183992"
    },
    "neoantigenutils_formatnetmhcpan - output(MUT,netmhcpanstab) - tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "typeMut": true,
                            "fromStab": true
                        },
                        "test_netmHCstabpanoutput.MUT.tsv:md5,246eb723691371ad49bd080071475740"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2c02f5f3103ee1532c27f7f3b873a578"
                ],
                "netMHCpanreformatted": [
                    [
                        {
                            "id": "test",
                            "typeMut": true,
                            "fromStab": true
                        },
                        "test_netmHCstabpanoutput.MUT.tsv:md5,246eb723691371ad49bd080071475740"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2c02f5f3103ee1532c27f7f3b873a578"
                ]
            }
        ],
        "timestamp": "2024-07-30T13:46:47.110076"
    },
    "neoantigenutils_formatnetmhcpan - output(WT,netmhcpanstab) - tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "typeMut": false,
                            "fromStab": false
                        },
                        "test_netmHCpanoutput.WT.tsv:md5,b95a6624d4010eb6517ca880a13e670d"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2c02f5f3103ee1532c27f7f3b873a578"
                ],
                "netMHCpanreformatted": [
                    [
                        {
                            "id": "test",
                            "typeMut": false,
                            "fromStab": false
                        },
                        "test_netmHCpanoutput.WT.tsv:md5,b95a6624d4010eb6517ca880a13e670d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2c02f5f3103ee1532c27f7f3b873a578"
                ]
            }
        ],
        "timestamp": "2024-07-30T13:46:56.841519"
    }
}